lib/src/util/js_component.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:over_react/over_react.dart';
+import 'package:react/react_client/component_factory.dart';
+
+/// Creates a Dart component factory that wraps a ReactJS [factoryProxy].
+///
+/// Example:
+/// ```dart
+/// UiFactory<ButtonProps> Button = uiJsComponent(
+///   ReactJsComponentFactoryProxy(MaterialUI.Button),
+///   _$ButtonConfig, // ignore: undefined_identifier
+/// );
+///
+/// @Props(keyNamespace: '')
+/// mixin ButtonProps on UiProps {}
+/// ```
+UiFactory<TProps> uiJsComponent<TProps extends UiProps>(
```

Comment:
Don't forget to create an entrypoint that exports this file and `prop_conversion,dart`. Otherwise, cocnsumers won't be able to access these APIs because they're under src.

We probably want a separate entrypoint from `over_react.dart`, so that once those APIs get exported they don't cause collision with the existing APIs defined in RMUI. How about something like `lib/js_component.dart`?

Reply:
done :)